custom_components/localtuya/__init__.py

```diff
@@ -246,29 +169,86 @@ async def _async_reconnect(now):
-    """Set up LocalTuya integration from a config entry."""
-    unsub_listener = entry.add_update_listener(update_listener)
+async def async_migrate_entry(hass, config_entry: ConfigEntry):
+    """Migrate old entries merging all of them in one."""
+    new_version = 2
+    if config_entry.version == 1:
+        _LOGGER.debug("Migrating config entry from version %s", config_entry.version)
+        stored_entries = hass.config_entries.async_entries(DOMAIN)
 
-    device = TuyaDevice(hass, entry.data)
+        if config_entry.entry_id == stored_entries[0].entry_id:
+            _LOGGER.debug(
+                "Migrating the first config entry (%s)", config_entry.entry_id
+            )
+            new_data = {}
+            new_data[CONF_REGION] = "eu"
+            new_data[CONF_CLIENT_ID] = "xxx"
+            new_data[CONF_CLIENT_SECRET] = "xxx"
+            new_data[CONF_USER_ID] = "xxx"
+            new_data[CONF_USERNAME] = DOMAIN
+            new_data[CONF_DEVICES] = {
+                config_entry.data[CONF_DEVICE_ID]: config_entry.data.copy()
+            }
+            new_data[ATTR_UPDATED_AT] = str(int(time.time() * 1000))
+            config_entry.version = new_version
+            hass.config_entries.async_update_entry(
+                config_entry, title=DOMAIN, data=new_data
+            )
+        else:
+            _LOGGER.debug(
+                "Merging the config entry %s into the main one", config_entry.entry_id
+            )
+            new_data = stored_entries[0].data.copy()
+            new_data[CONF_DEVICES].update(
+                {config_entry.data[CONF_DEVICE_ID]: config_entry.data.copy()}
+            )
+            new_data[ATTR_UPDATED_AT] = str(int(time.time() * 1000))
+            hass.config_entries.async_update_entry(stored_entries[0], data=new_data)
+            await hass.config_entries.async_remove(config_entry.entry_id)
+
+    _LOGGER.info(
+        "Entry %s successfully migrated to version %s.",
+        config_entry.entry_id,
+        new_version,
+    )
+    return True
 
-    hass.data[DOMAIN][entry.entry_id] = {
-        UNSUB_LISTENER: unsub_listener,
-        TUYA_DEVICE: device,
-    }
 
-    async def setup_entities():
-        platforms = set(entity[CONF_PLATFORM] for entity in entry.data[CONF_ENTITIES])
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up LocalTuya integration from a config entry."""
+    unsub_listener = entry.add_update_listener(update_listener)
+    hass.data[DOMAIN][UNSUB_LISTENER] = unsub_listener
+    hass.data[DOMAIN][TUYA_DEVICES] = {}
+
+    region = entry.data[CONF_REGION]
+    client_id = entry.data[CONF_CLIENT_ID]
+    secret = entry.data[CONF_CLIENT_SECRET]
+    user_id = entry.data[CONF_USER_ID]
+    tuya_api = TuyaCloudApi(hass, region, client_id, secret, user_id)
+    res = await tuya_api.async_get_access_token()
+    if res != "ok":
+        _LOGGER.error("Cloud API connection failed: %s", res)
```

Comment:
Ditto re not yet configured warnings - this is possibly a bit misleading.

```
Logger: custom_components.localtuya
Source: custom_components/localtuya/__init__.py:241
Integration: LocalTuya ([documentation](https://github.com/rospogrigio/localtuya/), [issues](https://github.com/rospogrigio/localtuya/issues))
First occurred: 9:06:18 PM (1 occurrences)
Last logged: 9:06:18 PM

Cloud API connection failed: Error 1114: your ip(149.167.167.219) don't have access to this API
```

Reply:
I know, but that message is coming from the Cloud, I am just forwarding it. This is what it usually replies if you input the wrong credentials, the reply is weird but I would keep it like this.